src/prpy/base/robot.py

```diff
@@ -288,107 +286,94 @@ def PostProcessPath(self, path, defer=False, executor=None,
             logger.debug('Detected "%s" tag on trajectory: Setting smooth'
                          ' = True', Tags.SMOOTH)
 
-        def do_postprocess():
-            with Clone(self.GetEnv()) as cloned_env:
-                cloned_robot = cloned_env.Cloned(self)
-
-                # Planners only operate on the active DOFs. We'll set any DOFs
-                # in the trajectory as active.
-                env = path.GetEnv()
-                cspec = path.GetConfigurationSpecification()
-
-                used_bodies = cspec.ExtractUsedBodies(env)
-                if self not in used_bodies:
-                    raise ValueError(
-                        'Robot "{:s}" is not in the trajectory.'.format(
-                            self.GetName()))
-
-                # Extract active DOFs from teh trajectory and set them as active.
-                dof_indices, _ = cspec.ExtractUsedIndices(self)
-
-                if util.HasAffineDOFs(cspec):
-                    affine_dofs = (DOFAffine.X | DOFAffine.Y | DOFAffine.RotationAxis)
-                    
-                    # Bug in OpenRAVE ExtractUsedIndices function makes 
-                    # dof_indices = affine_dofs. Temporary workaround for that bug.
-                    dof_indices = []
-                    logger.warning(
-                        'Trajectory contains affine DOFs. Any regular DOFs'
-                        ' will be ignored.'
-                    )
-                else:
-                    affine_dofs = 0
+        # Lazily create an environment for post-processing.
+        # This is faster than creating a new environment for every operation.
+        if not hasattr(self, '_postprocess_env'):
+            self._postprocess_env = openravepy.Environment()
 
-                cloned_robot.SetActiveDOFs(dof_indices, affine_dofs)
-                logger.debug(
-                    'Setting robot "%s" DOFs %s (affine? %d) as active for'
-                    ' post-processing.',
-                    cloned_robot.GetName(), list(dof_indices), affine_dofs
-                )
+        with Clone(self.GetEnv(),
+                   clone_env=self._postprocess_env) as cloned_env:
+            cloned_robot = cloned_env.Cloned(self)
```

Comment:
In fact, I don't see how we could do this otherwise, we need to set active DOFs on the robot before we call the retimers.  How do we do that without either locking the `env` for the duration of the `PostProcessPath()` call or using a clone?